WordPress/Classes/Models/BasePost.m

```diff
@@ -181,7 +182,7 @@ - (void)setDateCreated:(NSDate *)localDate
      If the date is now or in the past, and the status is scheduled, set the status
      to published.
      */
-    if (self.date_created_gmt == nil) {
+    if (self.date_created_gmt == nil || [self.date_created_gmt isEqualToDate:self.dateModified]) {
```

Comment:
PS: by the description of the PR it sounds to me like we should be checking if the post is a draft here too?